typhon/suite.py

```diff
@@ -326,6 +304,29 @@ def from_device(cls, device, parent=None,
                 display.add_tool(name, tool())
             except Exception:
                 logger.exception("Unable to load %s", type(tool))
-        param = display.add_device(device, **kwargs)
-        display.show_subdisplay(param)
+        display.add_device(device, **kwargs)
+        display.show_subdisplay(device)
         return display
+
+    def _add_to_sidebar(self, parameter, category=None):
+        """Add an item to the sidebar, connecting neccesary signals"""
+        if category:
+            # Create or grab our category
+            group_dict = dict((param.name(), param)
+                              for param in self.top_level_groups)
```

Comment:
Seeing this, maybe the above suggestion should be `get_top_level_group_dict()['Tools']` or something along those lines... 🚲 🏠

Reply:
Maybe `top_level_groups` just returns the dictionary. You can always strip it down to `values()` for the same affect.